src/sinks/azure_blob.rs

```diff
@@ -0,0 +1,832 @@
+        let container_name = self.container_name.as_str();
+        let client =
+            StorageAccountClient::new_connection_string(new_http_client(), connection_string)?
+                .as_storage_client()
+                .as_container_client(container_name);
+
+        Ok(client)
+    }
+}
+
+impl Service<AzureBlobSinkRequest> for AzureBlobSink {
+    type Response = PutBlockBlobResponse;
+    type Error = Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>;
+    type Future = BoxFuture<'static, StdResult<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<StdResult<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, request: AzureBlobSinkRequest) -> Self::Future {
+        let client =
+            Arc::<ContainerClient>::clone(&self.client).as_blob_client(request.blob_name.as_str());
+
+        Box::pin(async move {
+            let byte_size = request.blob_data.len();
+            let blob = client
+                .put_block_blob(Bytes::from(request.blob_data))
+                .content_type(request.content_type.unwrap());
+            let blob = match request.content_encoding {
+                Some(encoding) => blob.content_encoding(encoding),
+                None => blob,
+            };
+
+            blob.execute()
+                .inspect_err(|reason| {
+                    match reason.downcast_ref::<HttpError>() {
+                        Some(HttpError::UnexpectedStatusCode {
+                            expected: _,
+                            received,
+                            body: _,
+                        }) => {
+                            emit!(AzureBlobErrorResponse { code: *received })
+                        }
+                        _ => emit!(AzureBlobHttpError {
+                            error: reason.to_string()
+                        }),
+                    };
+                })
+                .inspect_ok(|result| {
+                    emit!(AzureBlobEventSent {
+                        request_id: result.request_id,
+                        byte_size
+                    })
+                })
+                .instrument(info_span!("request"))
+                .await
+        })
+    }
+}
+
+impl Compression {
+    pub fn content_type(&self) -> Option<&'static str> {
+        match self {
+            Self::None => Some("text/plain"),
+            Self::Gzip(_) => Some("application/octet-stream"),
+        }
+    }
+}
+
+impl Response for PutBlockBlobResponse {}
+
+impl RetryLogic for AzureBlobRetryLogic {
+    type Error = HttpError;
+    type Response = PutBlockBlobResponse;
+
+    fn is_retriable_error(&self, error: &Self::Error) -> bool {
+        match error {
+            HttpError::UnexpectedStatusCode {
+                expected: _,
+                received,
+                body: _,
+            } => received.is_server_error() || received == &StatusCode::TOO_MANY_REQUESTS,
```

Comment:
You can elide the unused parameters with `..`:
```suggestion
            HttpError::UnexpectedStatusCode { received, .. } =>
                received.is_server_error() || received == &StatusCode::TOO_MANY_REQUESTS,
```